### PR TITLE
Add default format and secondary format

### LINKED ISSUE #255: Add secondary format and default format to language rules
When switching to a new language, one should not have to set format explicitly, but have format for the given language already set, like for Java Gerrish gum bow, for Python snake laws etc.

Second format convention exists in most languages, like ClassName in Java or Python, there should be a command for that

### PR DESCRIPTION
**Pull request by comodoro**
closes #255

### CODE REVIEW

**Comment by LexiconCode** [comment]
For end-user configurable settings we should move out of the source code into settings.py 
```
    text_format = 3, 1 #gerrishGum
    secondary_text_format = 2, 1 #TieGum
```

**Comment by comodoro** [comment]
I did have doubts if this should be explicitly configurable. A dictionary in default settings to be loaded from ccrmerger.merge?

**Comment by LexiconCode** [comment]
> 
> 
> I did have doubts if this should be explicitly configurable. A dictionary in default settings to be loaded from ccrmerger.merge?

I agree ccrmerger.merge would be reasonable.

**Comment by LexiconCode** [comment]
@comodoro The last thing to do would be to update the documentation to reflect your commands.

**Comment by comodoro** [comment]
@LexiconCode I cannot find the appropriate place with related commands. I should probably create it in CCR.MD, under the Command sets paragraph, right?

**Comment by LexiconCode** [comment]
> 
> 
> @LexiconCode I cannot find the appropriate place with related commands. I should probably create it in CCR.MD, under the Command sets paragraph, right?

I would include it in the CasterQuickReference under text formatting with a new subcategory.

**Comment by comodoro** [comment]
OK, but it will not be immediately. The docx format is a bit unfortunate, I have tried Libre Office and Wordpad, but they both render it differently and both a bit wrong. I do not have MS Word at hand.

**Comment by comodoro** [comment]
@LexiconCode Realizing that this does not reflect 1. restarting Caster, 2. disabling grammars. I will add an appropriate data structure to settings to track last enabled grammars?

**Comment by comodoro** [comment]
Sorry I'm not clear. This does not change format from default if you restart Caster but enabled grammars are persisted. After disabling a grammar it does not revert to the last enabled grammar.

**Comment by comodoro** [comment]
Pushed the docs, feel free to correct me.

**Comment by LexiconCode** [comment]
@comodoro ready to merge?

**Comment by comodoro** [comment]
@LexiconCode I should make it match enabled grammars. I am pleasantly surprised that the toml thing does not require me to do anything.

**Comment by LexiconCode** [comment]
Thank you for this. We can re-factor at a later date as needed. Feel free to add in unit tests as that's important and something I know very little about.  Due to the size of the pull request I need at least one more person the test or review besides us before I am comfortable merging.

File: caster/lib/dfplus/merge/ccrmerger.py [hunk 1]
```diff
@@ -299,7 +332,7 @@ def merge(self, time, name=None, enable=True, save=False):
             rule.set_context(context)
             active_apps.append(rule)
         '''negation context for appless version of base rule'''
-        contexts = [rule.get_context() for rule in self._app_rules.values() \
+        contexts = [app_rule.get_context() for app_rule in self._app_rules.values() \
                     if rule.get_context() is not None]# get all contexts
```
Comment (mrob95) [inline caster/lib/dfplus/merge/ccrmerger.py]: Shouldn't this be?:
```suggestion
                    if app_rule.get_context() is not None]# get all contexts
```
or am I missing something?

**Comment by comodoro** [comment]
Thanks @mrob95 ! This was an oversight and confirms the rule "If it ain't broken, don't fix it". The reason for this was Python 2's leaking listcomp variables, but unnecessary and introducing bug on my part

**Comment by comodoro** [comment]
Added what I understand to be basic unit tests, E.g. without dependencies and very simple

**Comment by mrob95** [comment]
Looks good, I'd be happy to merge. I don't mind updating the docs in one go, probably more efficient than making individual changes with every PR.

**Comment by LexiconCode** [comment]
Alright so the plan is to merge and then add the docs in a second pull request?

**Pull request merged**